src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkYetUndefinedNameInspection.kt

```diff
@@ -18,16 +21,21 @@ class SmkYetUndefinedNameInspection : SnakemakeInspection() {
             session: LocalInspectionToolSession
     ) = object : SnakemakeInspectionVisitor(holder, session) {
         override fun visitPyReferenceExpression(node: PyReferenceExpression?) {
-            if (node?.text !in INSPECTED_KEYWORDS) {
+            val parentRun = PsiTreeUtil.getParentOfType(node, SMKRuleRunParameter::class.java)
+            if (node?.text !in INSPECTED_KEYWORDS || parentRun != null) {
                 return
             }
 
             val parent = node!!.parent
             val resolvedNode = parent.reference?.resolve() // Either corresponding rule or checkpoint or null
 
-            if (resolvedNode != null && resolvedNode.textOffset > parent.textOffset) {
-                registerProblem(parent,SnakemakeBundle.message("INSP.NAME.undefined.name") +
-                                ": ${(node.parent as PyReferenceExpression).name}")
+            if (resolvedNode != null && resolvedNode.containingFile == node.containingFile &&
+                (resolvedNode.textOffset > parent.textOffset ||
+                 resolvedNode === PsiTreeUtil.getParentOfType(node, SMKRule::class.java))) {
+                val identifier = node.nextSibling.nextSibling
```

Comment:
as always it is too low-level api and not stable, use public interfaces, here is `PyReferenceExpression.nameElement`